manifests/init.pp

```diff
@@ -0,0 +1,161 @@
+# @summary Manage a MySQL/MariaDB server configuration.
+#
+# @param create_mysql_home
+#   Should the mysql_home be created?
+#
+# @param dbs
+#   Raw data to define mysql::db resources.
+#
+# @param manage_yumrepo
+#   Should the mysql Yumrepo be managed? Must also provide
+#   Hiera data for the $yumrepo param.
+#
+# @param mysql_gid
+#   Optionally define the GID for the mysql group (and PGID for the mysql user).
+#   Must also provide $mysql_uid. If both $mysql_gid and $mysql_uid are provided,
+#   the mysql user and group will be created as will various directories.
+#
+# @param mysql_groupname
+#   The name of the mysql group.
+#
+# @param mysql_home
+#   The home dir of the mysql user.
+#
+# @param mysql_logdir
+#   Location of MySQL log files.
+#
+# @param mysql_rundir
+#   Location of run directory (generally a sub-directory of /var/run/).
+#
+# @param mysql_uid
+#   Optionally define the UID of the mysql user. Must also provide $mysql_gid. If
+#   both $mysql_gid and $mysql_uid are provided, the mysql user and group will be
+#   created as will various directories.
+#
+# @param mysql_username
+#   Username of the mysql user.
+#
+# @param other_dependencies
+#   Other depencies to handle prior to other MySQL setup, specified as resources, e.g.:
+#     - "Mount['/var/lib/mysql']"
+#
+# @param yumrepo
+#   Raw params for a yumrepo resource from which to install MySQL/MariaDB.
+#
+# @example
+#   include profile_mysql_server
+#
+class profile_mysql_server (
+
+  Boolean           $create_mysql_home,
+  Hash              $dbs,
+  Boolean           $manage_yumrepo,
+  String            $mysql_groupname,
+  String            $mysql_home,
+  String            $mysql_logdir,
+  String            $mysql_rundir,
+  String            $mysql_username,
+  Array             $other_dependencies,
+  Hash              $yumrepo,
+  Optional[Integer] $mysql_gid,
+  Optional[Integer] $mysql_uid,
+
+) {
+
+  include ::mysql::server
+
+  if ($mysql_gid) and ($mysql_uid) {
+    # if both $mysql_gid and $mysql_uid, create various directories
+    # and manage user and group
+
+    $dir_defaults = {
+      before => Class['::mysql::server::install'],
+      ensure => directory,
+      group  => $mysql_gid,
+      owner  => $mysql_uid,
+    }
+
+    if $mysql_logdir {
+      file { $mysql_logdir:
+        * => $dir_defaults,
+      }
+    }
+    if $mysql_rundir {
```

Comment:
From the README it isn’t obvious this rundir won’t be created if we don’t supply uid/gid. That was a definitely an argument for assigning defaults to the ids.

Or just document in README that this is skipped if no ids provided.

Reply:
Looks like the Red Hat-provided mariadb-server RPM creates /run/mariadb, and /var/run links to /run. (But as with the other dirs, if you don't want to/can't rely on consistent UID/GID from the RPM install, then you'd want to specify UID and GID to ensure consistent user, group, and folders are created prior to the install.)

I've updated the README to highlight this better.